articles/introduction-to-built-in-data-structures-in-java/index.md

```diff
@@ -0,0 +1,304 @@
+```java
+list1.set(1, "Welcome"); // Replaces "World" with "Welcome" in list1
+// OUTPUT: list1 = ["Hello", "Welcome"]
+```
+
+##### Remove Elements
+
+In `Array`, removal of elements is not possible, since the memory allocated is fixed.
+
+In `ArrayList`, the removal of elements can be done using the `remove(int index)` method. On specifying the `index`, the `ArrayList` removes the element from the memory, and the remaining elements are moved to fill up space. Thus, it is more memory efficient.
+
+Example:
+
+```java
+list1.remove(0); // Removes "Hello" from list1
+// OUTPUT: list1 = ["Welcome"]
+```
+
+#### LinkedList
+
+Linked lists are linear data structures that are connected using pointers, and holds data. The memory allocation is not contiguous which makes it better than the use of Arrays.
+
+While creating linked lists in Java, we have to create a separate class for working and managing the pointers. A sample class demonstrating the creation of pointers in user-defined linked list is shown as below:
+
+```java
+class Node { // Node class containing a single pointer and data
+  Node next; // 'next' is used as pointer
+  int data; // 'data' is used for storing values in the node
+  // Constructor for initializing the variables
+  Node() { 
+    next = null;
+    data = 0;
+  }
+}
+```
+
+In the `java.util` package, we simplify the tasks of manual creation for memory handling, by using `LinkedList` data structure. It contains pre-defined methods for building a linked list data structure.
+
+Syntax for creating `LinkedList` data structure:
+
+```java
+import java.util.LinkedList; // Import package
+
+LinkedList<String> list2 = new LinkedList<String>(); // Create new object
+```
+
+##### Add Elements
+
+In the user-defined linked list data structure, for adding new data, we have to manually move the pointers and assign data. Whereas, in the `LinkedList` data structure, we make use of `add(Object)` or `add(int index, Object)` methods for adding new elements.
+
+Example:
+
+```java
+list2.add("Hello"); // Add "Hello" to the LinkedList list2
+// OUTPUT: ["Hello"]
+list2.add("World"); // Add "World" to the LinkedList list2
+// OUTPUT: ["Hello", "World"]
+list2.add(1, "Computer"); // Add "Computer" to the LinkedList list2 at index 1
+// OUTPUT: ["Hello", "Computer", "World"]
+```
+
+##### Update Elements
+
+To update elements in the user-defined linked list, we have to traverse the whole linked list from the "HEAD" pointer to update the particular data.
+
+`LinkedList` data structure abstracts the whole process with the `set(index, data)` method.
+
+Example:
+
+```java
+list2.set(0, "Welcome"); // Update the string at index 0 to "Welcome"
+// OUTPUT: ["Welcome", "Computer", "World"]
+```
+
+##### Remove Elements
+
+Similarly, for deletion of an element in a user-defined linked list is cumbersome, where the pointer of the previous element is pointed to the address of the next element. When working with huge data, working with pointers is not easy.
```

Comment:
Could you give a quick example of deletion of an element in a linkedList using the standard method (without using in-built data structures)